ci/src/cI_github_hooks.mli

```diff
@@ -1,80 +1,17 @@
-open CI_utils
-open! Result
-
-type t
-
-module Commit_state : sig
-  type t
-
-  val status : t -> Datakit_S.status_state option Lwt.t
-  val descr : t -> string option Lwt.t
-  val target_url : t -> Uri.t option Lwt.t
-end
+open Datakit_github
 
 module CI : sig
-  type t
-  val of_string : string -> t
+  type t = string list
   val circle_ci : t
   val datakit_ci : string -> t
 end
 
-module Commit : sig
-  type t
-
-  val hash : t -> string
-  val pp : t Fmt.t
-  val state : CI.t -> t -> Commit_state.t
-  val project : t -> CI_projectID.t
-end
-
-module PR : sig
-  type t
-
-  val id : t -> int
-  val head : t -> Commit.t
-  val title : t -> string
-  val project : t -> CI_projectID.t
-  val dump : t Fmt.t
-  val compare : t -> t -> int
-end
-
-module Ref : sig
-  type t
-
-  val project : t -> CI_projectID.t
-  val name : t -> Datakit_path.t
-  val head : t -> Commit.t
-  val dump : t Fmt.t
-  val compare : t -> t -> int
-end
-
-val connect : DK.t -> t
-
-val pr : t -> project_id:CI_projectID.t -> int -> PR.t option Lwt.t
-
-val set_state : t -> CI.t -> status:Datakit_S.status_state -> descr:string -> ?target_url:Uri.t -> Commit.t -> unit Lwt.t
-
-module Target : sig
-  type t = [ `PR of PR.t | `Ref of Ref.t ]
-
-  val head : t -> Commit.t
-end
-
 module Snapshot : sig
-  type t
```

Comment:
But the caching is already done in the [engine](https://github.com/docker/datakit/blob/cd060d151465e37fab8419674f960893efce1403/ci/src/cI_engine.ml#L84) right?